aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -750,6 +750,44 @@ void cpu_scatter_reduce_expanded_index(const Tensor& self, const Tensor& index,
   });
 }
 
+template <typename scalar_t>
+void cpu_gather_expanded_index_kernel(const Tensor& result, const Tensor& index, const Tensor& self) {
+  int64_t* index_data = index.data_ptr<int64_t>();
```

Comment:
Unrelated to this PR, but I was just wondering if there are any mid-term or long-term plans to also support `int32`-based indexing?

Reply:
I have no idea, need to ask the core maintainers. Right now, `gather` is LongTensor only.

Reply:
Indexing is almost always done in int64 all throughout the library. 
TorchInductor on CUDA fixes that by proving max bounds on the values of the strides and using int32 if possible.